virtualmachine/vsphere/vm.go

```diff
@@ -886,6 +895,30 @@ func getToolsRunningStatus(status string) bool {
 	return false
 }
 
+//getDisksInfo  returns the disks info of this VM.
+func getDisksInfo(vmMo mo.VirtualMachine) []DiskInfo {
+	var disksInfo []DiskInfo
+	devices := object.VirtualDeviceList(vmMo.Config.Hardware.Device)
+	for _, device := range devices {
+		if disk, ok := device.(*types.VirtualDisk); ok {
+			if c := devices.FindByKey(disk.ControllerKey); c != nil {
```

Comment:
```FindByKey``` method traverses the devices slices to find the objects that match the passed key. Store disk controller and key in ```map[int][VirtualDevice]``` and lookup for the key in map before using ```FindByKey``` to prevent the traversals again and again for same key.

Reply:
Reduce nesting.